test/e2e/nodepool_additionalTrustBundlePropagation_test.go

```diff
@@ -107,5 +110,71 @@ func (k *AdditionalTrustBundlePropagationTest) Run(t *testing.T, nodePool hyperv
 			e2eutil.WithInterval(10*time.Second), e2eutil.WithTimeout(20*time.Minute),
 		)
 
+		t.Logf("Updating hosted cluster by removing additional trust bundle.")
+		if err = e2eutil.UpdateObject(t, k.ctx, k.mgmtClient, k.hostedCluster, func(obj *hyperv1.HostedCluster) {
+			obj.Spec.AdditionalTrustBundle = nil
+		}); err != nil {
+			t.Fatalf("failed to update HostedCluster with additional trust bundle: %v", err)
+		}
+
+		// Ensure the control plane operator deployment is updated and no longer mounts the additional trust bundle configmap
+		controlPlaneOperatorDeployment := &appsv1.Deployment{
```

Comment:
nice! an additional good validation would be the AdditionalTrustBundle secret doesn't exist in the cp namespace and also the TrustedCABundleConfigMap doesn't have it in their data.